zerver/tests/test_thumbnail.py

```diff
@@ -110,12 +105,13 @@ def test_thumbnail_redirect_for_spectator(self) -> None:
 
             self.logout()
             response = self.client_get("/thumbnail", {"url": url[1:], "size": "full"})
-            self.assertEqual(response.status_code, 302)
+            self.assertEqual(response.status_code, 200)
 
         # Deny file access since rate limited
         with ratelimit_rule(86400, 0, domain="spectator_attachment_access_by_file"):
             response = self.client_get("/thumbnail", {"url": url[1:], "size": "full"})
-            self.assertEqual(response.status_code, 403)
+            self.assertEqual(response.status_code, 302)
+            self.assertTrue(response["Location"].startswith("/accounts/login/?next="))
```

Comment:
The comment `# Deny file access since rate limited` is wrong; not sure how best to patch this.

Reply:
It still does -- it just redirects to login when it's rate-limited.  This is the behaviour of `/user_uploads/` in general for anonymous users hitting the rate-limit, which we can decide to tweak elsewhere.